src/ray/object_manager/pull_manager.cc

```diff
@@ -162,6 +162,18 @@ void PullManager::UpdatePullsBasedOnAvailableMemory(size_t num_bytes_available)
     }
   }
 
+  // Check whether we are at memory capacity.
+  if (num_bytes_being_pulled_ >= num_bytes_available_) {
+    // We have at least as many pull requests queued than we do memory
+    // capacity.
+    at_memory_capacity_ = true;
+  } else {
+    // We can potentially serve another pull request. There may actually be
+    // more requests in the queue, since we do not serve requests until we have
+    // the object size information.
+    at_memory_capacity_ = false;
+  }
```

Comment:
We can simplify by not checking this. I think it's fine to spill if we don't have object capacity information yet.